src/util/scroll.js

```diff
@@ -121,14 +121,16 @@ function isNumber (v: any): boolean {
   return typeof v === 'number'
 }
 
-const hashStartsWithNumberRE = /^#\d/
+const selectorLooksLikeIdRE = /^#[^, ]+$/
```

Comment:
why any character but `,` and ` `?

Reply:
@posva good question - logic is that spaces aren't allowed in an HTML5 id and both a space or a comma would suggest a complex selector, e.g:
```
<div id="one">  <!-- getElementById('one') -->
  <div class="two"> <!-- querySelector('#one .two') --> </div>
  <div id="one/two"><!-- getElementById('one/two') --></div></div>
<div id="two"><!-- querySelector('#two, #one') --> </div>
```
If special characters or ids starting with a number are used in a complex selector (i.e. one that starts with a `#` but contains a comma or space) they would need to be properly escaped as if they were being used in CSS / querySelector. This is currently the case and continues to apply in the PR

Reply:
This also means the onus isn't on the router to attempt to escape characters that aren't intended to be escaped, instead it simply checks for a simple id query and passes everything else to querySelector with all the caveats. The idea being that you can pass the hash fragment from a url or `to.hash` without parsing it. The only gotcha being that it can't contain a `,` (which is technically allowed in an id)

Reply:
@posva a `,` or ` ` suggests a complex selector (either nested or an OR). Technically an id selector could contain an `,` but the above provides the best balance of simplicity vs. coverage without breaking changes. Consider:
```
// Simple selector (passed to getElementById):
\#one
/#1
\#one/two

// Complex selector (passed to querySelector): 
\#one .two
\#one, .two
\#one\,two // selects id="one,two"
```
Did you have any suggestions?

Reply:
Hi @posva any update on this PR?